src/applets/clock/ClockApplet.vala

```diff
@@ -322,37 +290,40 @@ public class ClockApplet : Budgie.Applet {
 		}
 
 		// Prevent unnecessary redraws
-		var old = date_label.get_label();
-		var ctime = time.format(ftime);
+		var old = this.seconds_label.get_label();
+		var ctime = this.time.format(ftime);
 		if (old == ctime) {
 			return;
 		}
 
-		seconds_label.set_markup(ctime);
+		this.seconds_label.set_markup(ctime);
 	}
 
 	/**
 	 * This is called once every second, updating the time
 	 */
 	protected bool update_clock() {
-		time = new DateTime.now_local();
+		this.time = new DateTime.now(this.clock_timezone);
+		
 		string format;
+		if (!this.clock_use_custom_format) {
+			if (!this.clock_use_24_hour_time) {
+				format = "%l:%M";
+			} else {
+				format = "%H:%M";
+			}
 
-
-		if (ampm) {
-			format = "%l:%M";
-		} else {
-			format = "%H:%M";
-		}
-
-		if (orient == Gtk.Orientation.HORIZONTAL) {
-			if (check_seconds.get_active()) {
-				format += ":%S";
+			if (orient == Gtk.Orientation.HORIZONTAL) {
+				if (this.clock_show_seconds) {
```

Comment:
Could just be: `if ((orient == Gtk.Orientation.HORIZONTAL) && this.clock_show_seconds)`